workshops/hcload/README.md

```diff
@@ -0,0 +1,465 @@
+
+export default async function (): Promise<void> {
+        const app = new Application()
+        app.use(async (context: any) => {
+            context.response.body = "Server running"
+        })
+
+        app.addEventListener("listen", async ({ port }) => {
+            console.log("HTTP server ready")
+        })
+
+        await app.listen({ port: 20685 })
+}
+```
+
+At the moment, the server simply replies with "Server running" when anything is requested. If you trying running the program (simply run hcload), you should see HTTP server ready, and on the top left of your screen "Server running"
+
+- Next, let's ask for a filePath in our library's default function, and serve the file when an HTTP request reaches our server
+
+```js
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+
+export default async function (filePath: string): Promise<void> { // <--- Changed
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' }) // <--- Changed
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        console.log("HTTP server ready")
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Modify `hcload.ts` to pass a filePath to the library. Currently, you can pass the `LICENSE` file included with the starter code.
+
+```js
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+import hcload from "./mod.ts"
+
+hcload(path.resolve("./LICENSE"))
+```
+
+Run `hcload` and you should see the contents of the file `LICENSE` in your browser preview. Congratulations on making it so far!
+
+### Part 3.2: Exposing localhost via ngrok
+
+At the moment, files served will only be accessible over your home network (yes, over repl.it they'll be accessible over the internet, but the end use of this program is to upload files from your own computer). Since the Hackclub CDN needs to be able to make a request to your server over the internet, we'll need to expose the server over the internet.
+
+- Import the "ngrok" module and expose the server to the web (in `mod.ts`)
+
+```js
+import { connect, disconnect } from 'https://deno.land/x/ngrok@2.2.3/mod.ts'
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+
+export default async function (filePath: string): Promise<void> {
+    const fileBase = path.parse(filePath).base
+
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' })
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        const ngrokUrl = `https://${await connect({ protocol: 'http', port })}/${fileBase}`
+        console.log("NGROK: " + ngrokUrl)
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Here's what fileBase is:
+    - fileBase is the last part of the path to the file, e.g. For "/home/runner/myFile.png", the fileBase is "myFile.png"
```

Comment:
This renders weirdly in the HTML. Also if you're explaining what fileBase is, you don't need to prefix it with "Here's what fileBase is"